### PR TITLE
(BB-1359) Add translations configuration and automation

### PR DESCRIPTION
**Pull request by xirdneh**
This PR adds necessary configuration files to create translations. Working with translations is automated by using make targets which are explained in the README.

# Testing instructions:

1. Extract translations: `make extract_translations`
2. Build dummy translations: `make build_dummy_translations`
3. Validate translations: `make validate_translations`
4. Run lms, go to `/update_lang/` and update language to `eo`
5. Make sure strings are translated.

### CODE REVIEW

File: eoc_journal/translations/en/LC_MESSAGES/text.po-e [hunk 1]
```diff
@@ -0,0 +1,247 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
```
Comment (viadanna) [inline eoc_journal/translations/en/LC_MESSAGES/text.po-e]: Is this file required? Looks like a copy of the above.

Comment (xirdneh) [reply eoc_journal/translations/en/LC_MESSAGES/text.po-e]: @viadanna That's a backup file created by sed. I'll add it to `.gitignore`

File: .tx/config [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[main]
+host = https://www.transifex.com
+
+[xblocks.image-explorer]
```
[the remark this answers is not in the document]
Comment (xirdneh) [reply .tx/config]: @viadanna Ah, sorry about that. I've fixed this.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	cd $(WORKING_DIR) && i18n_tool extract
+	if [ -f "$(EXTRACTED_DJANGOJS)" ]; then msgcat $(EXTRACTED_DJANGO) $(EXTRACTED_DJANGOJS) -o $(EXTRACTED_TEXT); else mv $(EXTRACTED_DJANGO) $(EXTRACTED_TEXT); fi
+	rm -f $(EXTRACTED_DJANGO)
+	rm -f $(EXTRACTED_DJANGOJS)
+	sed -i'' -e 's/nplurals=INTEGER/nplurals=2/' $(EXTRACTED_TEXT)
+	sed -i'' -e 's/plural=EXPRESSION/plural=\(n != 1\)/' $(EXTRACTED_TEXT)
+
+compile_translations: ## compile translation files, outputting .mo files for each supported language
+	cd $(WORKING_DIR) && i18n_tool generate
+	python manage.py compilejsi18n --namespace EOCJournalXBlockI18N --output $(JS_TARGET)
```
Comment (viadanna) [inline Makefile]: :+1:

File: eoc_journal/settings.py [hunk 4]
```diff
@@ -77,7 +77,7 @@
 
 STATICI18N_NAMESPACE = 'EOCJournalXBlockI18N'
 STATICI18N_PACKAGES = (
-    'chat',
+    'eoc_journal',
```
Comment (viadanna) [inline eoc_journal/settings.py]: :+1:
